app/app.go

```diff
@@ -655,8 +656,13 @@ func New(
 	icaAuthModule := icaauth.NewAppModule(appCodec, app.ICAAuthKeeper)
 	icaAuthIBCModule := icaauth.NewIBCModule(app.ICAAuthKeeper)
 
-	icaControllerIBCModule := icacontroller.NewIBCMiddleware(icaAuthIBCModule, app.ICAControllerKeeper)
-	icaControllerStack := ibcfee.NewIBCMiddleware(icaControllerIBCModule, app.IBCFeeKeeper)
+	var icaControllerStack porttypes.IBCModule
+	icaControllerStack = icacontroller.NewIBCMiddleware(icaAuthIBCModule, app.ICAControllerKeeper)
+	icaControllerStack = ibcfee.NewIBCMiddleware(icaControllerStack, app.IBCFeeKeeper)
+	// Since the callbacks middleware itself is an ics4wrapper, it needs to be passed to the ica controller keeper
+	app.ICAControllerKeeper.WithICS4Wrapper(icaControllerStack.(porttypes.Middleware))
+	const maxCallbackGas = uint64(1000000)
```

Comment:
TODO, make it a consensus parameter in cronos module.

Reply:
Agree on the consensus param approach. We need discussion on its initial value, according to https://github.com/crypto-org-chain/cronos/pull/1178, the avg gas for IBC message is 300k, so allowing a callback to 1M gas sounds too me too huge.

I expect the smart contract should not do heavy computation in the callback, but perhaps just to flip a flag. Heavy computation should be done on a separate Tx call based on the updated flag value.